mlflow/protos/service.proto

```diff
@@ -2358,6 +2384,30 @@ message SetTraceTagV3 {
   reserved 1;reserved "request_id";
 }
 
+message SetTraceTagV4 {
+  option (scalapb.message).extends = "com.databricks.rpc.RPC[$this.Response]";
+
+  // ID of the trace on which to set a tag.
+  optional string trace_id = 1 [(validate_required) = true];
+
+  // Location of the trace. experiment_id or catalog.schema
+  optional string location = 2 [(validate_required) = true];
+
+  // Name of the tag. Maximum size depends on storage backend.
+  // All storage backends are guaranteed to support key values up to 250 bytes in size.
+  optional string key = 3 [(validate_required) = true];
+
+  // String value of the tag being logged. Maximum size depends on storage backend.
+  // All storage backends are guaranteed to support key values up to 250 bytes in size.
+  optional string value = 4;
+
+  // Optional SQL warehouse ID for setting trace tag.
+  optional string sql_warehouse_id = 5;
```

Comment:
I think we may not actually need the `sql_warehouse_id` currently here as this backend API currently does not need to read anything from delta, it just writes. However, we can bias toward keeping it in case we need to change this later, and see if anything comes up in API design review